sidechain/consensus/aura/src/block_importer.rs

```diff
@@ -159,6 +179,61 @@ impl<
 	pub(crate) fn block_author_is_self(&self, block_author: &SignedSidechainBlock::Public) -> bool {
 		self.authority.public() == *block_author
 	}
+
+	fn create_finish_game_extrinsic(
+		&self,
+		sidechain_block: &&<SignedSidechainBlock as SignedBlock>::Block,
+	) -> Result<(), ConsensusError> {
+		let calls = self
```

Comment:
What you're doing here is:
1) retrieve all calls from the pool (no matter if they just have been introduced or have been there for some time)
2) iterate through all of them
3) get state for all connect four play turn calls

That seems a bit over the top if you also query the state for calls that have not been included into the sidechain block yet and won't have affected the game state in any way. And might result in sending a finishing extrinsic multiple times for the same game.